### PR TITLE
#1415 optimize runtime compilation

### PR DESCRIPTION
**Pull request by mximp**
#1415 optimize runtime compilation as per: https://issues.apache.org/jira/browse/MCOMPILER-205

### CODE REVIEW

**Comment by yegor256** [comment]
@mximp did you try this instead (as suggested in Jira):

```
<compilerArgs>
<arg>-Xpkginfo:always</arg>
</compilerArgs>
```

or maybe `useIncrementalCompilation` (as it's suggested too)?

**Comment by mximp** [comment]
> @mximp did you try this instead (as suggested in Jira):
> 
> ```
> <compilerArgs>
> <arg>-Xpkginfo:always</arg>
> </compilerArgs>
> ```
> 
> or maybe `useIncrementalCompilation` (as it's suggested too)?

@yegor256 Fixed the former... The latter is enabled by default but it doesn't work (by design) as we need it.

File: eo-runtime/pom.xml [hunk 1]
```diff
@@ -89,6 +89,12 @@ SOFTWARE.
       <artifactId>junit-pioneer</artifactId>
       <!-- version from parent POM -->
     </dependency>
+    <dependency>
+      <groupId>org.eolang</groupId>
+      <artifactId>eo-maven-plugin</artifactId>
+      <version>${project.version}</version>
+      <scope>compile</scope>
```
Comment (yegor256) [inline eo-runtime/pom.xml:96]: @mximp I'm almost sure that if you put `provided` here, it will also work

File: pom.xml [hunk 2]
```diff
@@ -245,7 +245,11 @@ SOFTWARE.
         <plugin>
           <artifactId>maven-compiler-plugin</artifactId>
           <version>3.8.1</version>
-          <configuration combine.self="override"/>
+          <configuration combine.self="override">
+            <compilerArgs>
+              <arg>-Xpkginfo:always</arg>
```
Comment (yegor256) [inline pom.xml:250]: @mximp I think this will break `jacoco` -- coverage information will stop collecting

Comment (mximp) [reply pom.xml:250]: @yegor256 Why do you think so?

Comment (yegor256) [reply pom.xml:250]: @mximp my mistake, I mixed it up with this: https://github.com/objectionary/eo/blob/master/eo-runtime/pom.xml#L98 (surefire plugin, not compiler)

**Comment by mximp** [comment]
@yegor256 please check

File: eo-maven-plugin/src/main/java/org/eolang/maven/PackageMarker.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import java.lang.annotation.RetentionPolicy;
+
+/**
+ * Marker annotation for packages to make them compilable into `.class` file.
+ *
+ * @since 1.0
+ * @see <a href="https://issues.apache.org/jira/browse/MCOMPILER-205">MCOMPILER-205</a>
+ */
+@Retention(RetentionPolicy.SOURCE)
+public @interface PackageMarker {
```
Comment (yegor256) [inline eo-maven-plugin/src/main/java/org/eolang/maven/PackageMarker.java:36]: @mximp you never use this annotation. are you sure it's necessary?

Comment (mximp) [reply eo-maven-plugin/src/main/java/org/eolang/maven/PackageMarker.java:36]: @yegor256 definitely not.. I thought I removed it. Thanks!

**Comment by mximp** [comment]
@yegor256 please check

**Pull request merged**